src/metabase/channel/render/table.clj

```diff
@@ -98,46 +74,75 @@
          (when bar-width
            [:th {:style (style/style (bar-td-style) (bar-th-style) {:width (str bar-width "%")})}]))])
 
-(defn- render-bar
-  [bar-width normalized-zero]
-  (if (< bar-width normalized-zero)
-    (list
-     [:td {:style (style/style (bar-td-style) {:width :99%, :border-right "1px solid black", :padding-right 0})}
-      (render-bar-component (style/secondary-color)
-                            false
-                            (normalized-score->pixels (- normalized-zero bar-width))
-                            (normalized-score->pixels bar-width))]
-     [:td {:style (style/style (bar-td-style) {:width :99%})}])
-    (list
-     (when-not (zero? normalized-zero)
-       [:td {:style (style/style (bar-td-style) {:width :99%, :border-right "1px solid black"})}])
-     [:td {:style (style/style (bar-td-style) {:width :99%, :padding-left 0})}
-      (render-bar-component (style/primary-color)
-                            true
-                            (normalized-score->pixels (- bar-width normalized-zero)))])))
+(def ^:private minibar-width 100)
+
+(defn- render-minibar
+  "Return hiccup html structure for a numeric column cell with mini bar charts enabled. We use nested tables
+  as opposed to absolute and relative positioning for email client compatibility."
+  [val {:keys [min max]}]
+  (if-let [num (:num-value val)] ;; Assumes NumericWrapper
+    (let [is-neg?        (< num 1)
+          has-neg?       (< min 0)
+          normalized-max (clojure.core/max (abs min) (abs max))
+          pct-full       (int (* (/ (abs num) normalized-max) 100))
+          pct-left       (- minibar-width pct-full)
+          neg-pct-full   (int (Math/floor (* pct-full 0.49)))
+          neg-pct-left   (- 49 neg-pct-full)]
+      [:table {:style (style/style {:width "100px" :border "0" :border-collapse "collapse"})}
+       [:tr
+        [:td {:style (style/style {:width "30%" :vertical-align "middle" :padding-right "10px"})} (h val)]
+        [:td {:style (style/style {:width "70%" :vertical-align "middle" :padding "0"})}
+         [:table {:style (style/style {:width "100%" :border-collapse "collapse" :height "10px"})}
+          [:tr
+         ;; Minibars for series that contain negative values have a dividing center line representing 0
+         ;; with negative minibars extending to the left
+           (if has-neg?
+             [:tr
+              [:td {:style (style/style {:background-color (if is-neg? "#FCE1E1" "#E0ECF9") :border-radius "3px" :padding "0"})}
```

Comment:
I think it'd be worth pulling these hex codes out as vars with meaningful names. There are already a bunch of color definitions in `metabase.channel.render.style` but that's more about the email template rendering, not the data viz itself. But we could consider unifying with those or finding a new location for these on the rendering side